src/fileio/dataformat.cxx

```diff
@@ -48,7 +48,7 @@ void DataFormat::writeFieldAttributes(const std::string& name, const FieldPerp&
   int yindex = f.getIndex();
   if (yindex >= 0 and yindex < fieldmesh.LocalNy) {
     // write global y-index as attribute
-    setAttribute(name, "yindex_global", fieldmesh.getGlobalYIndex(f.getIndex()));
+    setAttribute(name, "yindex_global", fieldmesh.getGlobalYIndexNoBoundaries(yindex));
```

Comment:
see my earlier comment https://github.com/boutproject/BOUT-dev/pull/2102#discussion_r492238557
```suggestion
    setAttribute(name, "yindex_global", fieldmesh.getGlobalYIndex(yindex));
```

Reply:
I did try it this way, but it fails `test-io` because it ends up writing `fperp2` on processors 2 and 3, both with `yindex_global = 11`. 

I think maybe the conditional also needs changing to 

```cpp
if ((yindex >= fieldmesh.myg) and (yindex < fieldmesh.LocalNy - fieldmesh.myg)) {
```